src/rules/fileNameCasingRule.ts

```diff
@@ -28,6 +28,74 @@ enum Casing {
     SnakeCase = "snake-case",
 }
 
+type RegexConfig = Record<string, Casing>;
+
+type SimpleConfig = Casing;
```

Comment:
Readability matters, By using `SimpleConfig` in the source code we know what we are referring to. I prefer to bind `Casing` to `SimpleConfig` that having `RegexConfig` (notice the naming convention of `Config` and `Casing`.

P.S: I used to have it with just `Casing`

Reply:
> if you really feel strongly about it

I do, because it was confusing reading through it and seeing multiple types referring to the same and trying to wrap my head around why there was a different before realizing there wasn't any difference 😅.

In general, TSLint rules try to favor simplicity in their implementations. If there's no reason for added complexity (extra type definitions or aliases, etc.), it shouldn't be added.

Reply:
Similarly, would you mind also removing `ValidationResult`? It's just another level of misdirection that makes it harder to parse through later lines of code.

Reply:
As I commented on the other comment. The type systems is being used to understand the source code without merging contexts.

When we talk about the possible configurations we would use `SimpleConfig` and when we talk about the enum we use `Casing`.

I don't see any issue at all by casting one type into another one, specially when it helps to understand the types and their usage.